### PR TITLE
Update yaml examples for MemoryDB

### PR DESCRIPTION
**Pull request by kyriechen96**
Issue #, if available:
Yaml examples for MemoryDB is incomplete and have some mistakes.

Description of changes:
Create more yaml examples to cover most of user cases.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: resources/memorydb/v1alpha1/cluster/cluster_create_with_resource_names.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: memorydb.services.k8s.aws/v1alpha1
+kind: Cluster
+metadata:
+  name: $CLUSTER_NAME
+spec:
+  aclName: $ACL_NAME
+  autoMinorVersionUpgrade: true/false
+  description: $Description
```
Comment (RedbackThomson) [inline resources/memorydb/v1alpha1/cluster/cluster_create_with_resource_names.yaml]:
```suggestion
description: $DESCRIPTION
```

File: resources/memorydb/v1alpha1/cluster/cluster_restore_with_snapshot_name.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+apiVersion: memorydb.services.k8s.aws/v1alpha1
+kind: Cluster
+metadata:
+  name: $CLUSTER_NAME
+spec:
+  aclName: $ACL_NAME
+  autoMinorVersionUpgrade: true/false
+  description: $Description
```
Comment (RedbackThomson) [inline resources/memorydb/v1alpha1/cluster/cluster_restore_with_snapshot_name.yaml]:
```suggestion
description: $DESCRIPTION
```

File: resources/memorydb/v1alpha1/cluster/cluster_restore_with_snapshot_reference.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+apiVersion: memorydb.services.k8s.aws/v1alpha1
+kind: Cluster
+metadata:
+  name: $CLUSTER_NAME
+spec:
+  aclName: $ACL_NAME
+  autoMinorVersionUpgrade: true/false
+  description: $Description
```
Comment (RedbackThomson) [inline resources/memorydb/v1alpha1/cluster/cluster_restore_with_snapshot_reference.yaml]:
```suggestion
description: $DESCRIPTION
```

**Review by RedbackThomson — APPROVED** [review]
These look really really good. Just a few small nitpicks for the casing

File: resources/memorydb/v1alpha1/cluster/cluster_create_with_resource_names.yaml (hunk 1 above)
Comment (kyriechen96) [reply resources/memorydb/v1alpha1/cluster/cluster_create_with_resource_names.yaml]: Fixed.

**Review by RedbackThomson — APPROVED** [review]
/lgtm

**Pull request merged**